sdk/eventhub/event-hubs/test/sender.spec.ts

```diff
@@ -198,6 +198,77 @@ describe("EventHub Sender #RunnableInBrowser", function(): void {
     });
   });
 
+  describe("Create batch", function(): void {
+    it("should be sent successfully", async function(): Promise<void> {
+      const producer = client.createProducer({ partitionId: "0" });
+      const totalMessages = 3;
+      const eventDataBatch = await producer.createBatch();
+      for (let i = 1; i <= totalMessages; i++) {
+        const messageStatus = eventDataBatch.tryAdd({ body: `${Buffer.from("Z".repeat(650000))}` });
+        if (!messageStatus) {
+          break;
+        }
+      }
+      debug(`${eventDataBatch.events.length} messages sent in total.`);
+      should.equal(
+        eventDataBatch.events.length !== totalMessages,
+        true,
+        "Client was not able to send any messages"
+      );
+      await producer.send(eventDataBatch);
+    });
+
+    it("with partition key should be sent successfully.", async function(): Promise<void> {
+      const producer = client.createProducer({ partitionId: "0" });
+      const eventDataBatch = await producer.createBatch({ partitionKey: 1 as any });
+      for (let i = 0; i < 5; i++) {
+        eventDataBatch.tryAdd({ body: `Hello World ${i}` });
+      }
+      await producer.send(eventDataBatch);
+    });
+
+    it("with max message size should be sent successfully.", async function(): Promise<void> {
+      const producer = client.createProducer({ partitionId: "0" });
+      const eventDataBatch = await producer.createBatch({ maxMessageSize: 4096 });
+      for (let i = 0; i < 5; i++) {
+        eventDataBatch.tryAdd({ body: `Hello World ${i}` });
+      }
+      await producer.send(eventDataBatch);
+    });
+
+    it("should throw when maxMessageSize is greater than maximum message size on the AMQP sender link", async function(): Promise<
+    void
+  > {
+    try {
+      const producer = client.createProducer({ partitionId: "0" });
+      await producer.createBatch({ maxMessageSize: 2046528 });
+      throw new Error("Test Failure");
+    } catch (err) {
+      err.message.should.equal(
+        "Max message size is greater than maximum message size on the AMQP sender link."
+      );
+    }
+  });
+
+    it("should throw when Partition key is different than the one provided in the send options", async function(): Promise<
+      void
+    > {
+      try {
+        const producer = client.createProducer({ partitionId: "0" });
```

Comment:
Same comment here, I'd expect this to throw an error since `partitionId` is specified.